daemon/rpc_connect.go

```diff
@@ -51,8 +50,7 @@ func (r *RPC) connectWithContext(in *pb.ConnectRequest, srv pb.Daemon_ConnectSer
 
 	// set connection status to "Disconnected"
 	if didFail || err != nil {
-		r.connectionInfo.SetStatus(state.ConnectionStatus{State: pb.ConnectionState_DISCONNECTED, StartTime: nil})
-		r.vpnEvents.Connected.Publish(events.DataConnect{EventStatus: events.StatusFailure})
+		r.vpnEvents.Disconnected.Publish(events.DataDisconnect{EventStatus: events.StatusFailure})
```

Comment:
Maybe the `EventStatus` is not needed here? The `DataDisconnect` event is ignored anyway in `ConnectionStatusNotifyDisconnect`. Now it may be confusing, because it looks like you are saying that the Disconnect failed :smile: , previously it was `DataConnect` with failure

Reply:
I think it's fine with publishing `DataDisconnect` :thinking: , the details why we have disconnected state are not used anywhere anyway. We can extend this in the future if this will be needed